package/samplers/pfns4bo/sampler.py

```diff
@@ -0,0 +1,274 @@
+
+        self._device = utils.default_device
+
+        if isinstance(prior, torch.nn.Module):
+            trained_model = prior
+        elif prior == "vanilla gp":
+            _, _, trained_model, _ = train(**get_vanilla_gp_config(self._device))
+        elif prior == "hebo":
+            _, _, trained_model, _ = train(**get_heboplus_config(self._device))
+        else:
+            raise ValueError("You should specify `prior` as 'vanilla gp', 'hebo', or a model.")
+
+        self._model = trained_model
+        self._model.eval()
+
+        if model_path is not None:
+            torch.save(trained_model, model_path)
+
+    def sample_relative(
+        self, study: Study, trial: Trial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, Any]:
+        self._raise_error_if_multi_objective(study)
+
+        if search_space == {}:
+            return {}
+
+        states = (TrialState.COMPLETE,)
+        trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+
+        if len(trials) < self._n_startup_trials:
+            return {}
+
+        (
+            internal_search_space,
+            normalized_params,
+        ) = gp_search_space.get_search_space_and_normalized_params(trials, search_space)
+
+        _sign = -1.0 if study.direction == StudyDirection.MINIMIZE else 1.0
+        score_vals = np.array([_sign * cast(float, trial.value) for trial in trials])
+
+        if np.any(~np.isfinite(score_vals)):
+            warnings.warn(
+                "This sampler cannot handle infinite values. "
+                "We clamp those values to worst/best finite value."
+            )
+
+            finite_score_vals = score_vals[np.isfinite(score_vals)]
+            best_finite_score = np.max(finite_score_vals, initial=0.0)
+            worst_finite_score = np.min(finite_score_vals, initial=0.0)
+
+            score_vals = np.clip(score_vals, worst_finite_score, best_finite_score)
+
+        standarized_score_vals = (score_vals - score_vals.mean()) / max(1e-10, score_vals.std())
+
+        def rand_sample_func(n: int) -> torch.Tensor:
+            xs = sample_normalized_params(n, internal_search_space, None)
+            ret = torch.from_numpy(xs).to(torch.float32).to(self._device)
+            return ret
+
+        known_x = torch.from_numpy(normalized_params).to(torch.float32).to(self._device)
+        known_y = torch.from_numpy(standarized_score_vals).to(torch.float32).to(self._device)
+
+        with torch.enable_grad():
+            _, x_options, eis, _, _ = optimize_acq_w_lbfgs(
+                model=self._model,
+                known_x=known_x,
+                known_y=known_y,
+                num_grad_steps=self._num_grad_steps,
+                num_candidates=self._num_candidates,
+                pre_sample_size=self._pre_sample_size,
+                device=self._device,
+                rand_sample_func=rand_sample_func,
+                apply_power_transform=False,
```

Comment:
for us, power_transform was important to get really good scores. Maybe you can make it a hyper-parameter at least.

Reply:
Thank you for your input. I change the default value of it True.